test/hexlink.ts

```diff
@@ -183,4 +199,112 @@ describe("Hexlink", function() {
       )
     ).to.be.revertedWith("ERC1167: create2 failed");
   });
+
+  it("should deploy account with erc4337 entrypoint", async function() {
+    const { deployer } = await ethers.getNamedSigners();
+    const entrypoint = await ethers.getContractAt(
+      "EntryPoint",
+      (await deployments.get("EntryPoint")).address
+    );
+    expect(await ethers.provider.getCode(sender)).to.eq("0x");
+    // deposit eth before account created
+    await deployer.sendTransaction({
+      to: sender,
+      value: ethers.utils.parseEther("1.0")
+    });
+
+    const accountInitData = await buildAccountInitData(deployer.address)
+    const proof = await buildAuthProof(
+      hre,
+      senderNameHash,
+      accountInitData,
+      "validator",
+      hexlink.address
+    );
+    const initData = hexlink.interface.encodeFunctionData(
+      "deploy", [senderName, accountInitData, proof]
+    );
+    const initCode = ethers.utils.solidityPack(
+      ["address", "bytes"],
+      [hexlink.address, initData]
+    );
+
+    // build user op to deploy and send eth
+    const api = new SimpleAccountAPI({
+      provider: hre.ethers.provider,
+      entryPointAddress: entrypoint.address,
+      accountAddress: sender,
+      owner: deployer,
+      factoryAddress: hexlink.address,
+    });
+
+    const fee = await ethers.provider.getFeeData();
+    const callData = await buildAccountExecData(
+      deployer.address,
+      ethers.utils.parseEther("0.5"),
+    );
+
+    const userOp: UserOperationStruct = {
+      sender,
+      nonce: 0,
+      initCode,
+      callData,
+      callGasLimit: 500000,
+      verificationGasLimit: 2000000,
+      preVerificationGas: 0,
+      maxFeePerGas: fee.maxFeePerGas?.toNumber() ?? 0,
+      maxPriorityFeePerGas: fee.maxPriorityFeePerGas?.toNumber() ?? 0,
+      paymasterAndData: [],
+      signature: [],
+    };
+    const op = await resolveProperties(userOp);
+    const opHash = ethers.utils.keccak256(
+      ethers.utils.defaultAbiCoder.encode(
+        [
+          'address',
+          'uint256',
+          'bytes32',
+          'bytes32',
+          'uint256',
+          'uint256',
+          'uint256',
+          'uint256',
+          'uint256',
+          'bytes32',
+        ],
+        [
+          op.sender,
+          op.nonce,
+          ethers.utils.keccak256(op.initCode),
```

Comment:
If they changed the hash fn here. My have two points:
1. We are using createSignedOp from BaseAccountAPI supplied by 4337, which should include this change?
2. otherwise they forgot to change BaseAccountAPI since it did not be exposed until my recent change, so no one find this issue?

Reply:
we need to use 0.6.0 instead of 0.5.0. They tagged a ver0.6.0 branch at the github repo but never released the npm packages. We need to copy/paste the latest version to our own repo before using it in our repo

Reply:
yea, last time u mentioned this point, but I just stopped digging deeper because sdk 0.6 has not been released. But this is ridiculous since they changed downstream first A.K.A entrypoint which totally does not make sense TBH

Reply:
agree but this is life : )